hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -411,15 +413,15 @@ public void flush() {
     public ICompletableFuture<V> getAsync(K key) {
         checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
 
-        return getAsyncInternal(key);
+        return new DelegatingFuture<V>(getAsyncInternal(key), getSerializationService());
```

Comment:
Does this mean that we get an additional object per invocation? If so, then I'm not in favor of merging this pr.

Reply:
fixed.